sdk/keyvault/azsecrets/client.go

```diff
@@ -67,12 +50,21 @@ func NewClient(vaultUrl string, credential azcore.TokenCredential, options *Clie
 		options = &ClientOptions{}
 	}
 
-	conn := internal.NewConnection(credential, options.toConnectionOptions())
+	options.PerRetryPolicies = append(
```

Comment:
We shouldn't mutate options passed by applications (what happens when an application passes the same options pointer in multiple calls)